controllers/sessionController.js

```diff
@@ -0,0 +1,49 @@
+const User = require("../models/User");
+const parseVErr = require("../utils/parseValidationErrs");
+
+const registerShow = (req, res) => {
+  res.render("register");
+};
+
+const registerDo = async (req, res, next) => {
+  if (req.body.password != req.body.password1) {
+    req.flash("error", "The passwords entered do not match.");
+    return res.render("register", {  errors: flash("errors") });
+  }
+  try {
+    await User.create(req.body);
+  } catch (e) {
+    if (e.constructor.name === "ValidationError") {
+      parseVErr(e, req);
+    } else if (e.name === "MongoServerError" && e.code === 11000) {
+      req.flash("error", "That email address is already registered.");
+    } else {
+      return next(e);
+    }
+    return res.render("register", {  errors: flash("errors") });
```

Comment:
Requires change:
This line should be using `req.flash()` instead of `flash()`. Also, it should be `req.flash("error")` instead of  `req.flash("errors")` (no “s”). This will be true for ALL error messages!